### ISSUE TITLE
Update service command to be based on app routes

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#342]
`minishift service` command is not working ATM. See https://github.com/minishift/minishift/issues/79#issuecomment-275987353.

```
$ oc whoami
developer

$ oc status
In project demo on server https://192.168.42.243:8443
http://nodejs-ex-demo.192.168.42.243.xip.io to pod port 8080-tcp (svc/nodejs-ex)
  dc/nodejs-ex deploys istag/nodejs-ex:latest <-
    bc/nodejs-ex source builds https://github.com/openshift/nodejs-ex on openshift/nodejs:4 
    deployment #1 deployed 5 minutes ago - 1 pod
3 warnings identified, use 'oc status -v' to see details.

$ oc get pods
NAME                READY     STATUS      RESTARTS   AGE
nodejs-ex-1-btptw   1/1       Running     0          5m
nodejs-ex-1-build   0/1       Completed   0          8m

$ oc expose svc/nodejs-ex
route "nodejs-ex" exposed

$ oc get routes
NAME        HOST/PORT                              PATH      SERVICES    PORT       TERMINATION
nodejs-ex   nodejs-ex-demo.192.168.42.243.xip.io             nodejs-ex   8080-tcp   

$ oc get endpoints nodejs-ex  
NAME        ENDPOINTS         AGE
nodejs-ex   172.17.0.2:8080   9m

$ oc get svc
NAME        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
nodejs-ex   172.30.16.186   <none>        8080/TCP   8m

$ minishift service nodejs-ex -n demo 
Service demo/nodejs-ex does not have a node port. To assign a port automatically, the service type must be NodePort or LoadBalancer, but this service is of type ClusterIP.

$ minishift service nodejs-ex --url -n demo           
Service demo/nodejs-ex does not have a node port. To assign a port automatically, the service type must be NodePort or LoadBalancer, but this service is of type ClusterIP.
```

**Comment by hferentschik** [comment]
I think this is partly due to how this is implemented. The service command is using the Kube API to to this, but we want to do this via oc. There is also  mismatch I think what this command is supposed to report, since I think it is actually working, just not what one might expect.

Bottom line, this command needs to switch to use _oc_. This will also allow us to get rid of the Kubernetes dependency then. See also issue #168.

**Comment by budhrg** [comment]
This command seems to be working well but needs some documentation changes.

```
$ minishift service -h
Prints the URL for the specified service to the console.

Usage:
  minishift service [flags] SERVICE
  minishift service [command]

Available Commands:
  list        Gets the URLs of the services in your local cluster.

Flags:
      --format string      The URL format of the service. (default "http://{{.IP}}:{{.Port}}")
      --https              Access the service with HTTPS instead of HTTP.
  -n, --namespace string   The namespace of the service. (default "default")
      --url                Access the service in the command-line console instead of the default browser.
$ oc expose dc nodejs-ex 
service "nodejs-ex" exposed

$ oc get dc
NAME        REVISION   DESIRED   CURRENT   TRIGGERED BY
nodejs-ex   1          1         1         config,image(nodejs-ex:latest)

$ oc expose dc nodejs-ex --type=NodePort
service "nodejs-ex" exposed

$ oc get svc
NAME        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
nodejs-ex   172.30.159.2   <nodes>       8080/TCP   1m

$ minishift service nodejs-ex -n myproject --url
http://192.168.99.100:31550

$ curl -I http://192.168.99.100:31550
HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: text/html; charset=utf-8
Content-Length: 40382
ETag: W/"9dbe-U7JADjtt3b5U9eoIGU/lMGU7AHQ"
Date: Thu, 02 Mar 2017 09:06:59 GMT
Connection: keep-alive

$ minishift service list -n myproject
|-----------|-----------|-----------------------------|
| NAMSEPACE |   NAME    |             URL             |
|-----------|-----------|-----------------------------|
| myproject | nodejs-ex | http://192.168.99.100:31550 |
|-----------|-----------|-----------------------------|
```

@hferentschik For me it seems that we need documentation changes here. WDYT ?

**Comment by hferentschik** [comment]
@budhrg so what are you actually saying here?

**Comment by budhrg** [comment]
After discussion, we come into conclusion that this should be based on https://github.com/minishift/minishift/issues/465.

**Comment by budhrg** [comment]
After discussion, we come into new findings.

- `minikube service <app-name>` actual open the application `app-name` in the browser.
   Here, in terms of minishift, we want to open the app route in browser.
- `minikube service list` actually show list of available URL alongwith namespace, app name and URL
  Here, for minishift, we can show namespace, app name and route. One more column will be added to show `NodePort` if it is set.

**Comment by LalatenduMohanty** [comment]
> minikube service <app-name> actual open the application app-name in the browser.
> Here, in terms of minishift, we want to open the app route in browser.

@budhrg @hferentschik We should allow users to run minishift in a remote machine (through SSH) and there are command line browsers too. I am not against opening the route in browser. For example ``minikube service <app-name>` can list route in CLI and `minikube service <app-name> --console`  should open in browser.

**Comment by budhrg** [comment]
> For example ``minikube service can list route in CLI andminikube service --console` should open in browser.

It seems as per `minkube`, it was `minishift service appname` will open application in browser and `minishift service appname --url` will print the url.

**Comment by LalatenduMohanty** [comment]
@budhrg ok, if we are keeping flags from minikube supported in `minishift openshift service` we should be fine.
```
Flags:
      --format string      Format to output service URL in.  This format will be applied to each url individually and they will be printed one at a time. (default "http://{{.IP}}:{{.Port}}")
      --https              Open the service URL with https instead of http
  -n, --namespace string   The service namespace (default "default")
      --url                Display the kubernetes service URL in the CLI instead of opening it in the default browser

```

**Comment by hferentschik** [comment]
I don't think that `--format` and `--https` makes so much sense to us. I think we should drop these two flags. Especially flags does not make sense.

**Comment by hferentschik** [comment]
> I am not against opening the route in browser. For example minishift service <app-name> can list route in CLI and minishift service <app-name> --console should open in browser.

You do this via the '--url' flag, the same as `minishift console --url`, the equivalent is `minishift openshift service foo --url`

> For example minishift service <app-name> can list route in CLI and minishift service <app-name> --console should open in browser.

-1 To be consistent, we should keep the existing semantics.

**Comment by LalatenduMohanty** [comment]
+1 for existing semantics, I was not sure if we are keeping it. I need to look more about `--format` and `--https` before agreeing or disagreeing on whether we need it or not.

**Comment by budhrg** [comment]
> namespace is hardcode here - https://github.com/minishift/minishift/blob/master/cmd/minishift/cmd/openshift/service.go#L88 to default.

I think we can change it to myproject. WDYT ?

If not, yes you are right and the namespace needs to be specified. If this is the case we should dynamically determine the current namespace and apply it (in a follow up issue)
Yes, better to track in separate issue.

**Comment by gbraad** [comment]
I believe `--https` still makes sense to keep!
`oc expose` allows you to setup secured routes.

**Comment by budhrg** [comment]
I think we can keep the flag `--https` since its user choice whether he want to use it or not.

> oc expose allows you to setup secured routes.

If so, then sure we need it otherwise there is not way user can access secured route until he do `oc get route` himself.

**Comment by hferentschik** [comment]
Merged via pull request #597
